igvm/hypervisor_preferences.py

```diff
@@ -22,23 +22,28 @@
 
 class InsufficientResource(object):
     """Check a resource of hypervisor would be sufficient"""
-    def __init__(self, attribute, reserved=0):
-        self.attribute = attribute
+    def __init__(self, hv_attribute, vm_attribute, reserved=0):
+        self.hv_attribute = hv_attribute
+        self.vm_attribute = vm_attribute
         self.reserved = reserved
 
     def __repr__(self):
-        args = repr(self.attribute)
+        args = repr(self.hv_attribute)
         if self.reserved:
             args += ', reserved=' + repr(self.reserved)
 
         return '{}({})'.format(type(self).__name__, args)
 
     def __call__(self, vm, hv):
-        total_size = hv.dataset_obj[self.attribute]
-        vms_size = sum(v[self.attribute] for v in hv.dataset_obj['vms'])
+        # Treat freshly created HVs always passing this check
+        if not hv.dataset_obj[self.hv_attribute]:
+            return False
```

Comment:
Why pass and not fail here? You'll always have enough proper once with graphite data anyway and if you don't, this isn't a nock-out criteria anyway. Failing this by returning True seems to me the more safe thing to do, no?

Reply:
A freshly installed HV should be usable immediately, not after a few hours only when cron job on Serveradmin runs.

Reply:
Hmm. Fair enough.

Reply:
It would be usable but less likey be choosen as you return False here no ?

Reply:
This is not for sorting preference but for having HV in result or not having it at all. Function is called "InsufficientResource" and False means that the resource is not insufficient.